### ISSUE TITLE
TypeScript / JavaScript support

### ISSUE THREAD

**Issue by aholmes** [CodeConnect/AliveFeedback#1]
It would be great to have support for these languages.

**Comment by JoshVarty** [comment]
Alive runs user code by running your unit tests. It helps us guarantee that all program state is properly taken care of.

What sort of testing frameworks exist for Javascript/Typescript that you'd like to see support for?

Supporting Javascript and Typescript is definitely possible and reasonable.

**Comment by aholmes** [comment]
I have used http://jasmine.github.io/ for testing JavaScript. I haven't used any specifically for TypeScript, though there is this project https://github.com/Steve-Fenton/tsUnit.

**Comment by agauniyal** [comment]
@JoshVarty  Can we have c++ support too?

**Comment by JoshVarty** [comment]
@agauniyal Unfortunately compilation in C/C++ is prohibitively expensive at this time. It's not that we don't want to build Alive for C/C++ but that we can't :(

**Comment by agauniyal** [comment]
@JoshVarty yup I understand. If you guys ever want ( in near future ) to hop into c++ bandwagon , take a look at clang and clangtools. Plus there's a Cling made over clang too which was demonstrated in recent talks. These are just suggestions :) and best of luck for your product!

**Comment by tejacques** [comment]
[Mocha](http://mochajs.org/) is another popular JS/TS test runner.

**Comment by JoshVarty** [comment]
@tejacques Awesome, thanks I'll check it out!

**Comment by D1no** [comment]
[Velocity](https://github.com/meteor-velocity/velocity), the reactive test-runner for [Meteor](https://www.meteor.com/), the only JavaScript full stack framework that resides as Nr. 10 on Github's [most stared repos list](https://github.com/search?p=1&q=stars%3A%3E1&s=stars&type=Repositories).

**Comment by JoshVarty** [comment]
Thanks for the additions @D1no!

**Comment by TomasKaz** [comment]
:+1: for JS/TS. We've using Jasmine too.

**Comment by bcherny** [comment]
another :+1: also using jasmine + typescript/javascript/coffeescript
